### PR TITLE
Mock tests

### PR DESCRIPTION
**Pull request by mullenlucas**
## Implemented features

- Created the Solver class
- Implemented the `factorial` method for calculating the factorial result of a number from the parameter
- Implemented the `reverse` method for reversing a string
- Implemented the method `fizzbuzz` that takes an integer argument and print 'fizz', 'buzz' or 'fizzbuzz' according to the divisible properties of that integer argument.
- Implemented tests for the Solver class and all their methods

### CODE REVIEW

File: mock.rb [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
-    end
-  end
\ No newline at end of file
+describe Solver do
+  it 'finds the factorial of 5' do
+    calculator = Solver.new
+    expect(calculator.factorial(5)).to eq(120)
+  end
+end
+
```
Comment (Mihndim2020) [inline mock.rb:12]: - [x] Good job with test coverage, but since we are aiming for 100% test coverage, it will be nice to add tests for a negative number, zero and one. Please kindly add tests for the factorial of a negative number, zero  and one.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi Team, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION



# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: mock.rb (hunk 1 above)
Comment (mullenlucas) [reply mock.rb:12]: Thank you for your observation ! Indeed it is a good practice to test limit cases, so will do !

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
